ipie/addons/ephqmc/trial_wavefunction/toyozawa.py

```diff
@@ -0,0 +1,99 @@
+import numpy as np
+import scipy.linalg
+
+from ipie.addons.eph.trial_wavefunction.coherent_state import CoherentStateTrial
+from ipie.utils.backend import arraylib as xp
+from ipie.addons.eph.trial_wavefunction.variational.toyozawa_variational import circ_perm
+
+class ToyozawaTrial(CoherentStateTrial):
+    """"""
+    def __init__(self, wavefunction, hamiltonian, num_elec, num_basis, verbose=False):
+        super().__init__(wavefunction, hamiltonian, num_elec, num_basis, verbose=verbose)
+        self.perms = list(circ_perm([i for i in range(self.nbasis)]))
+        self.nperms = len(self.perms)
+
+    def calculate_energy(self, system, hamiltonian):
+        #TODO variational_energy_coherent_state in ipie.estimators.local_energy
+        pass
+
+    def calc_overlap(self, walkers) -> np.ndarray:
+        _ = self.calc_phonon_overlap(walkers)
+        _ = self.calc_electronic_overlap(walkers)
+        walkers.total_ovlp = walkers.el_ovlp * walkers.ph_ovlp 
+        walkers.ovlp = np.sum(walkers.total_ovlp, axis=1)
+        return walkers.ovlp
+
+
+    def calc_phonon_overlap(self, walkers) -> np.ndarray:
+        """"""
+        ph_ovlp = np.zeros(walkers.nwalkers, dtype=np.complex128)
+        for ip,perm in enumerate(self.perms):
+            ph_ov = np.exp(-(self.m * self.w0 / 2) * (walkers.x - self.beta_shift[perm])**2)
+            walkers.ph_ovlp[:, ip] = np.prod(ph_ov, axis=1)
+        ph_ovlp = np.sum(walkers.ph_ovlp, axis=1)  
+        return ph_ovlp
+
+    def calc_phonon_gradient(self, walkers) -> np.ndarray:  
+        r"""No reevaluation of phonon overlap because it reuses the overlap from the previous
```

Comment:
Docstrings should have a single descriptive line followed by a more detailed description if necessary.